web/src/tippyjs.js

```diff
@@ -647,6 +649,24 @@ export function initialize() {
             instance.destroy();
         },
     });
+
+    delegate("body", {
+        target: "#subscription_overlay .stream_settings_header .cannot-subscribe-tooltip",
+        appendTo: () => document.body,
+        onShow(instance) {
+            const stream_id = stream_settings_ui.get_active_data().id;
+            const stream_name = sub_store.get(stream_id).name;
+            const content = $t(
+                {defaultMessage: "Cannot subscribe to private stream #{stream_name}"},
```

Comment:
I am not sure how difficult it would be but can we replace `#` with the lock icon here?